openj9.test.daa/src/test.daa/net/openj9/test/arithmetics/TestValidityChecking.java

```diff
@@ -748,6 +749,490 @@ public void testEmptyArrayForCheckPacked()
 
     }
 
+    @Test
+    public void testFailingArguments()
+    {
+        final byte[] a = new byte[] {(byte) 0x40, (byte) 0xF9, (byte) 0xF1, (byte) 0xF2, (byte) 0xC3};
+        int result = ExternalDecimal.checkExternalDecimal(a, 0, 5, DecimalData.EBCDIC_SIGN_EMBEDDED_TRAILING, 5);
+        Assert.assertEquals("Precision == bytesWithSpaces should result in return code of 3", 3, result);
+
+        result = ExternalDecimal.checkExternalDecimal(a, 0, 5, DecimalData.EBCDIC_SIGN_EMBEDDED_TRAILING, -5);
+        Assert.assertEquals("bytesWithSpaces < 0 should result in return code of 3", 3, result);
+
+        result = ExternalDecimal.checkExternalDecimal(a, 2, 5, DecimalData.EBCDIC_SIGN_EMBEDDED_TRAILING, 1);
+        Assert.assertEquals("bytesWithSpaces < 0 should result in return code of 3", 2, result);
+
+        try {
+            ExternalDecimal.checkExternalDecimal(a, 0, -5, DecimalData.EBCDIC_SIGN_EMBEDDED_TRAILING, 5);
+        }
+        catch (IllegalArgumentException e) {
+            e.getMessage().contains("Precision must be greater than 0");
```

Comment:
Assuming this is testing for IllegalArgumentException, shouldn't we pass or fail based on the result. Also `contains`  method call do return boolean. Shouldn't we compare it ?